nav2_util/include/nav2_util/node_utils.hpp

```diff
@@ -161,63 +161,88 @@ using NodeParamInterfacePtr = rclcpp::node_interfaces::NodeParametersInterface::
+  bool warn_if_no_override = false, bool strict_param_loading = false,
   const ParameterDescriptor & parameter_descriptor = ParameterDescriptor())
 {
   if (param_interface->has_parameter(parameter_name)) {
     rclcpp::Parameter param(parameter_name, default_value);
     param_interface->get_parameter(parameter_name, param);
     return param.get_value<ParamType>();
   }
-  if (warn_if_no_override && param_interface->get_parameter_overrides().find(parameter_name) ==
-    param_interface->get_parameter_overrides().end())
-  {
-    RCLCPP_WARN_STREAM(
-        logger,
-        "Failed to get param " << parameter_name << " from overrides, using default value.");
-  }
-  return param_interface
-         ->declare_parameter(parameter_name, rclcpp::ParameterValue{default_value},
+
+  auto return_value = param_interface
+    ->declare_parameter(parameter_name, rclcpp::ParameterValue{default_value},
       parameter_descriptor)
-         .get<ParamType>();
+    .get<ParamType>();
+
+  const bool no_param_override = param_interface->get_parameter_overrides().find(parameter_name) ==
+    param_interface->get_parameter_overrides().end();
+  if (no_param_override) {
+    if (warn_if_no_override) {
+      RCLCPP_WARN_STREAM(
+            logger,
+            "Failed to get param " << parameter_name << " from overrides, using default value.");
+    }
+    if (strict_param_loading) {
+      std::string description = "Parameter " + parameter_name +
+        " not in overrides and strict_param_loading is True";
+      throw rclcpp::exceptions::InvalidParameterValueException(description.c_str());
+    }
+  }
+
+  return return_value;
 }
 
 /// If the parameter is already declared, returns its value,
 /// otherwise declares it and returns the default value
 /**
  * If the parameter is already declared, returns its value,
  * otherwise declares it and returns the default value.
- * The method will by default print a warning when the override is missing.
+ * The method can be configured to print a warn message or throw an InvalidParameterValueException
+ * when the override is missing by enabling the parameters warn_on_missing_params
+ * or strict_param_loading respectively.
  *
  * \param[in] node Pointer to a node object
  * \param[in] parameter_name Name of the parameter
  * \param[in] default_value Default value of the parameter
- * \param[in] warn_if_no_override If true, prints a warning whenever the parameter override is missing
  * \param[in] parameter_descriptor Optional parameter descriptor
  * \return The value of the param from the override if existent, otherwise the default value
  */
 template<typename ParamType, typename NodeT>
 ParamType declare_or_get_parameter(
   NodeT node, const std::string & parameter_name,
-  const ParamType & default_value, bool warn_if_no_override = false,
+  const ParamType & default_value,
   const ParameterDescriptor & parameter_descriptor = ParameterDescriptor())
 {
+  bool warn_if_no_override{false};
+  try {
```

Comment:
I don't see any bugs, but why not have this be set as default value of `false` rather than the try/catch based on declarations based on type for `warn_if_no_override` and `strict_param_loading`?